File: 19-1.py
N = int(input())
kratn = N % 4
  
if kratn == 2:
  print(N + 1)
elif kratn == 3:
  print(0)
elif kratn == 0:
  print(-N)
elif kratn == 1:
  print(1)
else:
  print(1)

File: 19-1.pas
program p;
var N, kratn : integer;

begin
  read(N);
  kratn := N mod 4;
  
  if kratn = 2 then
    write(N + 1)
  else if kratn = 3 then
    write(0)
  else if kratn = 0 then
    write(-N)
  else if kratn = 1 then
    write(1)
  else
    write(1);  
end.